### PR TITLE
Fix clippy errors from 1.72

### PR DESCRIPTION
**Pull request by rylev**
This fixes clippy errors with the version of clippy shipped with 1.72

### CODE REVIEW

File: crates/app/src/metadata.rs
Comment (itowlson) [inline crates/app/src/metadata.rs:26]: Hmm, I don't much like this lint.  I see why it's correct but it's less clear what it's doing and why - if `key` ever changed to a non-`Copy` type, the maintainer would need to reconstruct the clone logic.  I suppose that is too trivial to worry about though.

My preference would be to exclude this lint and keep the code explicit.  But if other folks prefer the elided version then it's no biggie..

File: crates/core/src/host_component.rs
Comment (itowlson) [inline crates/core/src/host_component.rs:109]: Same comment - prefer to exclude the lint but not a blocker

**Review by itowlson — APPROVED** [review]
There's one of the lints where I'd prefer to exclude it keep the affected code as is, but it's not a blocker.

**Comment by rylev** [comment]
I don't really mind either version - the newer version is arguably a bit clearer to me, but I can see why the more verbose might be preferable. Honestly, given the toss up, I'd prefer to just merge this as is and not have to wait for another CI run.

**Pull request merged**
